sentry-android-replay/src/main/java/io/sentry/android/replay/ScreenshotRecorder.kt

```diff
@@ -78,26 +90,37 @@ internal class ScreenshotRecorder(
                     Log.e("BITMAP CAPTURED", bitmap.toString())
                     val viewHierarchy = bitmapToVH[bitmap]
 
-                    if (viewHierarchy != null) {
-                        val canvas = Canvas(bitmap)
+                    val scaledBitmap = Bitmap.createScaledBitmap(
```

Comment:
`l`: You could have the `if (viewHierarchy == null) {` check before creating the `scaledBitmap`, to save some unnecessary memory allocation